press/subscribers/purge_cache.py

```diff
@@ -18,39 +15,33 @@ def _gen_purge_url(base, ids):
     return '{}/content/{}/latest.*$'.format(base, regex_ids)
 
 
-def _make_request(url):
-    """requests.Request factory for creating a purge request"""
-    purge_req = requests.Request('PURGE_REGEXP', url)
-    return purge_req.prepare()
-
-
 # subscriber for press.events.LegacyPublicationFinished
 def purge_cache(event):
     logger = event.request.log
     just_ids = list(map(lambda x: x[0], event.ids))
 
+    # Get the celery task object
+    task_path = '.'.join([make_request.__module__, make_request.__name__])
+    _make_request = event.request.registry.celery_app.tasks[task_path]
+
     # Build the legacy domain from the current request domain.
     domain = convert_to_legacy_domain(event.request.domain)
     # Build the base part of the purge url
     scheme = event.request.scheme
     base_url = '{}://{}'.format(scheme, domain)
 
-    with requests.Session() as session:
-        start = 0
-        range_stop = len(just_ids) + ID_CHUNK_SIZE
-        for end in range(ID_CHUNK_SIZE, range_stop, ID_CHUNK_SIZE):
```

Comment:
The previous code was reusing the session object in a loop here. The newer code enqueues(?) a new task for each. Not sure if there are performance implications.

Reply:
I think there might be performance issues if there's only 1 worker etc. With multiple workers it might be faster.

Reply:
It's probably not critically important. The performance boost will be that this is happening outside of the main process. It the tasks become slow, we can address that at that time. But as you point out, easy solution to that might be to pull up more workers.